kedro/io/data_catalog.py

```diff
@@ -346,6 +356,13 @@ def _sort_patterns(cls, dataset_patterns: Patterns) -> dict[str, dict[str, Any]]
                 pattern,
             ),
         )
+        catch_all = [
+            pattern for pattern in sorted_keys if cls._specificity(pattern) == 0
+        ]
+        if len(catch_all) > 1:
+            raise DatasetError(
+                f"Multiple catch-all patterns found in the catalog: {', '.join(catch_all)}"
```

Comment:
It might be helpful to add here how to resolve the problem by making sure only one catch-all exists in the catalog.

Reply:
done